### ISSUE TITLE
MQTTTransportStage: _on_mqtt_disconnect called: ConnectionDroppedError('Paho returned rc==1')

### ISSUE THREAD

**Issue by cristianCarrerasCastillo** [Azure/azure-iot-sdk-python#810]
# Context
- **OS and version used:** 
  PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
  NAME="Raspbian GNU/Linux"
  VERSION_ID="10"
  VERSION="10 (buster)"
  VERSION_CODENAME=buster
  ID=raspbian
- **Python version:** Python 3.7.3
- **Pip Version:** pip 18.1 from /usr/lib/python3/dist-packages/pip (python 3.7)`
- **azure-sdk version:** azure-iot-device    2.7.1 
- **Pip list:**
```
Package             Version    
------------------- -----------
asn1crypto          0.24.0     
astroid             2.1.0      
asttokens           1.1.13     
automationhat       0.2.0      
azure-iot-device    2.7.1      
beautifulsoup4      4.7.1      
blinker             1.4        
blinkt              0.1.2      
buttonshim          0.0.2      
Cap1xxx             0.1.3      
certifi             2018.8.24  
chardet             3.0.4      
Click               7.0        
colorama            0.3.7      
colorzero           1.1        
configparser        3.5.0b2    
cookies             2.2.1      
cryptography        2.6.1      
cupshelpers         1.0        
deprecation         2.1.0      
docutils            0.14       
drumhat             0.1.0      
entrypoints         0.3        
enum34              1.1.6      
envirophat          1.0.0      
ExplorerHAT         0.4.2      
Flask               1.0.2      
fourletterphat      0.1.0      
gpiozero            1.6.2      
html5lib            1.0.1      
idna                2.6        
ipaddress           1.0.17     
isort               4.3.4      
itsdangerous        0.24       
janus               0.4.0      
jedi                0.13.2     
Jinja2              2.10       
keyring             17.1.1     
keyrings.alt        3.1.1      
lazy-object-proxy   1.3.1      
logilab-common      1.4.2      
lxml                4.3.2      
Markdown            3.0.1      
MarkupSafe          1.1.0      
mccabe              0.6.1      
microdotphat        0.2.1      
mote                0.0.4      
motephat            0.0.3      
mypy                0.670      
mypy-extensions     0.4.1      
numpy               1.16.2     
oauthlib            2.1.0      
olefile             0.46       
packaging           20.9       
paho-mqtt           1.5.1      
pantilthat          0.0.7      
parso               0.3.1      
pexpect             4.6.0      
pgzero              1.2        
phatbeat            0.1.1      
pianohat            0.1.0      
picamera            1.13       
piglow              1.2.5      
pigpio              1.78       
Pillow              5.4.1      
pip                 18.1       
psutil              5.5.1      
pycairo             1.16.2     
pycrypto            2.6.1      
pycups              1.9.73     
pygame              1.9.4.post1
Pygments            2.3.1      
PyGObject           3.30.4     
pyinotify           0.9.6      
PyJWT               1.7.0      
pylint              2.2.2      
pyOpenSSL           19.0.0     
pyparsing           2.4.7      
pyserial            3.4        
pysmbc              1.0.15.6   
PySocks             1.7.1      
python-apt          1.8.4.3    
pyxdg               0.25       
PyYAML              3.13       
rainbowhat          0.1.0      
reportlab           3.5.13     
requests            2.21.0     
requests-oauthlib   1.0.0      
requests-unixsocket 0.2.0      
responses           0.9.0      
roman               2.0.0      
RPi.GPIO            0.7.0      
RTIMULib            7.2.1      
scrollphat          0.0.7      
scrollphathd        1.2.1      
SecretStorage       2.3.1      
Send2Trash          1.5.0      
sense-hat           2.2.0      
setuptools          40.8.0     
simplejson          3.16.0     
six                 1.12.0     
skywriter           0.0.7      
smbus2              0.4.1      
sn3218              1.2.7      
soupsieve           1.8        
spidev              3.5        
ssh-import-id       5.7        
thonny              3.3.10     
touchphat           0.0.1      
twython             3.7.0      
typed-ast           1.3.1      
unicornhathd        0.0.4      
update              0.4.4      
urllib3             1.26.5     
webencodings        0.5.1      
Werkzeug            0.14.1     
wheel               0.32.3     
wrapt               1.10.11
```

## Description of the issue

when I send msg with .send_message (msg) the sdk fails but not immediately, it takes a couple of messages for it to start not sending them.
this function is being called approximately every 3 minutes

## Code sample exhibiting the issue

```
async def Send_data(name_file):
    '''
    Carga el archivo json y lo envia a azure.
    * retorna True: si es que se pudo enviar el msj
    * retorna False: Si es que no se pudo enviar el msj
    '''
    print('Funcion de envio')
    key = Read_Key()
    try:
        cliente = iothub_client_init(key) # IoTHubDeviceClient.create_from_connection_string(CONNECTION_STRING, connection_retry=False)
        print(':::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::')
        try:
            await cliente.connect()
            file_send = open(PATH_JSON + name_file, 'r') 
            print(':::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::')
            try:
                await cliente.send_message('[' + file_send.read() + ']')
                print('::::::::::::::::...Mensaje enviado...::::::::::::::::::::::::::::::::::::::::::::::::::::::::')
                file_send.close()
                #await asyncio.sleep(10)
                await cliente.shutdown()
                return True
            except:
                print("No se pudo enviar el mensaje")
                await asyncio.sleep(1)
                await cliente.shutdown()
                await cliente.disconnect()
        except:
            print('No se pudo conectar a Azure')
            await asyncio.sleep(1)
            await cliente.disconnect()
            return False

    except:
        print(decor.CRED + 'ERROR!.' + decor.CEND + decor.s
        + 'No se pudo enviar el mensaje a azure' + decor.s)
        return False
```

## Console log of the issue
```
INFO:azure.iot.device.iothub.aio.async_clients:Initiating client shutdown
INFO:azure.iot.device.iothub.aio.async_clients:Disconnecting from Hub...
DEBUG:azure.iot.device.iothub.aio.async_clients:Executing initial disconnect
DEBUG:azure.iot.device.iothub.pipeline.mqtt_pipeline:Starting DisconnectOperation on the pipeline
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting run_op in pipeline thread
INFO:azure.iot.device.common.pipeline.pipeline_stages_base:ReconnectStage(DisconnectOperation): State changes LOGICALLY_DISCONNECTED->LOGICALLY_DISCONNECTED.  Sending op down.
INFO:azure.iot.device.common.pipeline.pipeline_stages_base:ConnectionLockStage(DisconnectOperation): Transport is already disconnected.  Completing.
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:DisconnectOperation: completing without error
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting on_complete in callback thread
DEBUG:azure.iot.device.common.async_adapter:Callback completed with result None
DEBUG:azure.iot.device.iothub.aio.async_clients:Successfully executed initial disconnect
DEBUG:azure.iot.device.iothub.aio.async_clients:Stopping handlers...
DEBUG:azure.iot.device.iothub.aio.async_clients:Successfully stopped handlers
DEBUG:azure.iot.device.iothub.aio.async_clients:Executing secondary disconnect...
DEBUG:azure.iot.device.iothub.pipeline.mqtt_pipeline:Starting DisconnectOperation on the pipeline
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting run_op in pipeline thread
INFO:azure.iot.device.common.pipeline.pipeline_stages_base:ReconnectStage(DisconnectOperation): State changes LOGICALLY_DISCONNECTED->LOGICALLY_DISCONNECTED.  Sending op down.
INFO:azure.iot.device.common.pipeline.pipeline_stages_base:ConnectionLockStage(DisconnectOperation): Transport is already disconnected.  Completing.
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:DisconnectOperation: completing without error
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting on_complete in callback thread
DEBUG:azure.iot.device.common.async_adapter:Callback completed with result None
DEBUG:azure.iot.device.iothub.aio.async_clients:Successfully executed secondary disconnect
INFO:azure.iot.device.iothub.aio.async_clients:Successfully disconnected from Hub
DEBUG:azure.iot.device.iothub.aio.async_clients:Beginning pipeline shutdown operation
DEBUG:azure.iot.device.iothub.pipeline.mqtt_pipeline:Commencing shutdown of pipeline
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting run_op in pipeline thread
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:Cancelling SAS Token renewal alarm
INFO:azure.iot.device.common.mqtt_transport:Forcing paho disconnect to prevent it from automatically reconnecting
DEBUG:azure.iot.device.common.mqtt_transport:Done forcing paho disconnect
DEBUG:azure.iot.device.common.mqtt_transport:Cancelling all pending operations
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:ShutdownPipelineOperation: completing without error
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting on_complete in callback thread
DEBUG:azure.iot.device.common.async_adapter:Callback completed with result None
DEBUG:azure.iot.device.iothub.aio.async_clients:Completed pipeline shutdown operation
INFO:azure.iot.device.iothub.aio.async_clients:Client shutdown complete
INFO:azure.iot.device.iothub.aio.async_clients:Disconnecting from Hub...
DEBUG:azure.iot.device.iothub.aio.async_clients:Executing initial disconnect
INFO:azure.iot.device.iothub.aio.async_clients:Disconnecting from Hub...
DEBUG:azure.iot.device.iothub.aio.async_clients:Executing initial disconnect
DEBUG:asyncio:Using selector: EpollSelector
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting run_op in pipeline thread
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_http:Gateway Hostname not present. Setting Hostname to: G****************.net
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_http:HTTPTransportStage(InitializePipelineOperation): got connection args
DEBUG:azure.iot.device.common.http_transport:creating a SSL context
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:InitializePipelineOperation: completing without error
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting <azure.iot.device.common.evented_callback.EventedCallback object at 0xac2e2e10> in callback thread
DEBUG:azure.iot.device.common.evented_callback:Callback completed with result None
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting run_op in pipeline thread
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:Scheduling SAS Token renewal at epoch time: 1625162050
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_mqtt:Gateway Hostname not present. Setting Hostname to: Grinding-Molino01-SM-Pruebas-IoT.azure-devices.net
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_mqtt:MQTTTransportStage(InitializePipelineOperation): got connection args
DEBUG:azure.iot.device.common.mqtt_transport:creating mqtt client
INFO:azure.iot.device.common.mqtt_transport:Creating client for connecting using MQTT over TCP
DEBUG:azure.iot.device.common.mqtt_transport:creating a SSL context
DEBUG:azure.iot.device.common.mqtt_transport:configuring SSL context with default certs
DEBUG:azure.iot.device.common.mqtt_transport:Created MQTT protocol client, assigned callbacks
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:InitializePipelineOperation: completing without error
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting <azure.iot.device.common.evented_callback.EventedCallback object at 0xb0d29fd0> in callback thread
DEBUG:azure.iot.device.common.evented_callback:Callback completed with result None
INFO:azure.iot.device.iothub.aio.async_clients:Connecting to Hub...
DEBUG:azure.iot.device.iothub.pipeline.mqtt_pipeline:Starting ConnectOperation on the pipeline
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting run_op in pipeline thread
INFO:azure.iot.device.common.pipeline.pipeline_stages_base:ReconnectStage(ConnectOperation): State changes LOGICALLY_DISCONNECTED->LOGICALLY_CONNECTED.  Adding to wait list and sending new connect op down
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:ReconnectStage: sending new connect op down
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:ConnectionLockStage(ConnectOperation): blocking
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_mqtt:MQTTTransportStage(ConnectOperation): connecting
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_mqtt:MQTTTransportStage(ConnectOperation): Starting watchdog
DEBUG:azure.iot.device.common.mqtt_transport:connecting to mqtt broker
INFO:azure.iot.device.common.mqtt_transport:Connect using port 8883 (TCP)
DEBUG:paho:Sending CONNECT (u1, p1, wr0, wq0, wf0, c0, k60) client_id=b'gateway_py'
DEBUG:azure.iot.device.common.mqtt_transport:_mqtt_client.connect returned rc=0
DEBUG:paho:Received CONNACK (1, 0)
INFO:azure.iot.device.common.mqtt_transport:connected with result code: 0
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting _on_mqtt_connected in pipeline thread
INFO:azure.iot.device.common.pipeline.pipeline_stages_mqtt:_on_mqtt_connected called
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:PipelineRootStage: ConnectedEvent received. Calling on_connected_handler
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting _on_connected in callback thread
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_mqtt:completing connect op
INFO:azure.iot.device.iothub.abstract_clients:Connection State - Connected
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_mqtt:MQTTTransportStage(ConnectOperation): cancelling watchdog
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:ConnectOperation: completing without error
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:ConnectionLockStage(ConnectOperation): op succeeded.  Unblocking queue
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:ConnectionLockStage(ConnectOperation): unblocking and releasing queued ops.
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:ConnectionLockStage(ConnectOperation): processing 0 items in queue for error=None
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:ReconnectStage(ConnectOperation): on_connect_complete error=None state=LOGICALLY_CONNECTED never_connected=True connected=True 
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:ReconnectStage: completing waiting ops with error=None
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:ConnectOperation: completing without error
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting on_complete in callback thread
DEBUG:azure.iot.device.common.async_adapter:Callback completed with result None
INFO:azure.iot.device.iothub.aio.async_clients:Successfully connected to Hub
INFO:azure.iot.device.iothub.aio.async_clients:Sending message to Hub...
DEBUG:azure.iot.device.iothub.pipeline.mqtt_pipeline:Starting SendD2CMessageOperation on the pipeline
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting run_op in pipeline thread
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:SendD2CMessageOperation: creating worker op of type MQTTPublishOperation
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:AutoConnectStage(MQTTPublishOperation): Connected.  Sending down and adding callback to check result
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_mqtt:MQTTTransportStage(MQTTPublishOperation): publishing on devices/gateway_py/messages/events/
INFO:azure.iot.device.common.mqtt_transport:publishing on devices/gateway_py/messages/events/
DEBUG:paho:Sending PUBLISH (d0, q1, r0, m1), 'b'devices/gateway_py/messages/events/'', ... (175245 bytes)
DEBUG:azure.iot.device.common.mqtt_transport:_mqtt_client.publish returned rc=0
DEBUG:azure.iot.device.common.mqtt_transport:Waiting for response on MID: 1
INFO:azure.iot.device.common.mqtt_transport:disconnected with result code: 1
DEBUG:azure.iot.device.common.mqtt_transport:  File "/home/pi/.vscode-server/extensions/ms-python.python-2021.6.944021595/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydev_bundle/pydev_monkey.py", line 919, in __call__
    ret = self.original_func(*self.args, **self.kwargs)
  File "/usr/lib/python3.7/threading.py", line 885, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/home/pi/.local/lib/python3.7/site-packages/paho/mqtt/client.py", line 3452, in _thread_main
    self.loop_forever(retry_first_connection=True)
  File "/home/pi/.local/lib/python3.7/site-packages/paho/mqtt/client.py", line 1779, in loop_forever
    rc = self.loop(timeout, max_packets)
  File "/home/pi/.local/lib/python3.7/site-packages/paho/mqtt/client.py", line 1181, in loop
    rc = self.loop_read(max_packets)
  File "/home/pi/.local/lib/python3.7/site-packages/paho/mqtt/client.py", line 1574, in loop_read
    return self._loop_rc_handle(rc)
  File "/home/pi/.local/lib/python3.7/site-packages/paho/mqtt/client.py", line 2227, in _loop_rc_handle
    self._do_on_disconnect(rc, properties)
  File "/home/pi/.local/lib/python3.7/site-packages/paho/mqtt/client.py", line 3360, in _do_on_disconnect
    self.on_disconnect(self, self._userdata, rc)
  File "/home/pi/.local/lib/python3.7/site-packages/azure/iot/device/common/mqtt_transport.py", line 210, in on_disconnect
    logger.debug("".join(traceback.format_stack()))

INFO:azure.iot.device.common.mqtt_transport:Forcing paho disconnect to prevent it from automatically reconnecting
DEBUG:azure.iot.device.common.mqtt_transport:in paho thread.  nulling _thread
DEBUG:azure.iot.device.common.mqtt_transport:Done forcing paho disconnect
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting _on_mqtt_disconnected in pipeline thread
INFO:azure.iot.device.common.pipeline.pipeline_stages_mqtt:MQTTTransportStage: _on_mqtt_disconnect called: ConnectionDroppedError('Paho returned rc==1')
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:ReconnectStage(DisconnectedEvent): State is LOGICALLY_CONNECTED Connected is True.
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:ReconnectStage(DisconnectedEvent): Not attempting to reconnect (Reconnect disabled)
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:PipelineRootStage: DisconnectedEvent received. Calling on_disconnected_handler
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting _on_disconnected in callback thread
INFO:azure.iot.device.common.pipeline.pipeline_stages_mqtt:MQTTTransportStage: disconnection was unexpected
INFO:azure.iot.device.iothub.abstract_clients:Connection State - Disconnected
INFO:azure.iot.device.iothub.abstract_clients:Cleared all pending method requests due to disconnect
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_mqtt:MQTTTransportStage: Connection Retry disabled - cancelling in-flight operations
DEBUG:azure.iot.device.common.mqtt_transport:Cancelling all pending operations
DEBUG:azure.iot.device.common.mqtt_transport:Cancelling 1 - Triggering callback
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Already in pipeline thread for on_complete
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:MQTTPublishOperation: completing with error OperationCancelled('Operation cancelled before PUBACK received')
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:SendD2CMessageOperation: Worker op (MQTTPublishOperation) has been completed
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:SendD2CMessageOperation: completing with error OperationCancelled('Operation cancelled before PUBACK received')
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting on_complete in callback thread
INFO:azure.iot.device.common.handle_exceptions:Unexpected disconnection
INFO:azure.iot.device.common.async_adapter:Callback completed with error OperationCancelled('Operation cancelled before PUBACK received')
INFO:azure.iot.device.common.handle_exceptions:azure.iot.device.common.transport_exceptions.ConnectionDroppedError: ConnectionDroppedError('Paho returned rc==1')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/pi/.local/lib/python3.7/site-packages/azure/iot/device/common/handle_exceptions.py", line 43, in swallow_unraised_exception
    raise e
azure.iot.device.common.transport_exceptions.ConnectionDroppedError: ConnectionDroppedError(None) caused by ConnectionDroppedError('Paho returned rc==1')

INFO:azure.iot.device.common.async_adapter:["azure.iot.device.common.pipeline.pipeline_exceptions.OperationCancelled: OperationCancelled('Operation cancelled before PUBACK received')\n"]
INFO:azure.iot.device.iothub.aio.async_clients:Initiating client shutdown
INFO:azure.iot.device.iothub.aio.async_clients:Disconnecting from Hub...
DEBUG:azure.iot.device.iothub.aio.async_clients:Executing initial disconnect
DEBUG:azure.iot.device.iothub.pipeline.mqtt_pipeline:Starting DisconnectOperation on the pipeline
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting run_op in pipeline thread
INFO:azure.iot.device.common.pipeline.pipeline_stages_base:ReconnectStage(DisconnectOperation): State changes LOGICALLY_DISCONNECTED->LOGICALLY_DISCONNECTED.  Sending op down.
INFO:azure.iot.device.common.pipeline.pipeline_stages_base:ConnectionLockStage(DisconnectOperation): Transport is already disconnected.  Completing.
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:DisconnectOperation: completing without error
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting on_complete in callback thread
DEBUG:azure.iot.device.common.async_adapter:Callback completed with result None
DEBUG:azure.iot.device.iothub.aio.async_clients:Successfully executed initial disconnect
DEBUG:azure.iot.device.iothub.aio.async_clients:Stopping handlers...
DEBUG:azure.iot.device.iothub.aio.async_clients:Successfully stopped handlers
DEBUG:azure.iot.device.iothub.aio.async_clients:Executing secondary disconnect...
DEBUG:azure.iot.device.iothub.pipeline.mqtt_pipeline:Starting DisconnectOperation on the pipeline
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting run_op in pipeline thread
INFO:azure.iot.device.common.pipeline.pipeline_stages_base:ReconnectStage(DisconnectOperation): State changes LOGICALLY_DISCONNECTED->LOGICALLY_DISCONNECTED.  Sending op down.
INFO:azure.iot.device.common.pipeline.pipeline_stages_base:ConnectionLockStage(DisconnectOperation): Transport is already disconnected.  Completing.
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:DisconnectOperation: completing without error
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting on_complete in callback thread
DEBUG:azure.iot.device.common.async_adapter:Callback completed with result None
DEBUG:azure.iot.device.iothub.aio.async_clients:Successfully executed secondary disconnect
INFO:azure.iot.device.iothub.aio.async_clients:Successfully disconnected from Hub
DEBUG:azure.iot.device.iothub.aio.async_clients:Beginning pipeline shutdown operation
DEBUG:azure.iot.device.iothub.pipeline.mqtt_pipeline:Commencing shutdown of pipeline
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting run_op in pipeline thread
DEBUG:azure.iot.device.common.pipeline.pipeline_stages_base:Cancelling SAS Token renewal alarm
INFO:azure.iot.device.common.mqtt_transport:Forcing paho disconnect to prevent it from automatically reconnecting
DEBUG:azure.iot.device.common.mqtt_transport:Done forcing paho disconnect
DEBUG:azure.iot.device.common.mqtt_transport:Cancelling all pending operations
DEBUG:azure.iot.device.common.pipeline.pipeline_ops_base:ShutdownPipelineOperation: completing without error
DEBUG:azure.iot.device.common.pipeline.pipeline_thread:Starting on_complete in callback thread
DEBUG:azure.iot.device.common.async_adapter:Callback completed with result None
DEBUG:azure.iot.device.iothub.aio.async_clients:Completed pipeline shutdown operation
INFO:azure.iot.device.iothub.aio.async_clients:Client shutdown complete
INFO:azure.iot.device.iothub.aio.async_clients:Disconnecting from Hub...
DEBUG:azure.iot.device.iothub.aio.async_clients:Executing initial disconnect
INFO:azure.iot.device.iothub.aio.async_clients:Disconnecting from Hub...
DEBUG:azure.iot.device.iothub.aio.async_clients:Executing initial disconnect
```

**Comment by cristianCarrerasCastillo** [comment]
I ran the program today and can send msg without problem, but after a couple of minutes, I started with the same problem. I have the log of this but I do not know how to share it, since it is long

**Comment by cristianCarrerasCastillo** [comment]
yesterday I ran the program, sending the data to a production iothub (at first I was sending it to a free one) and I did not present this problem again

**Comment by cartertinney** [comment]
@cristianCarrerasCastillo I am glad things are working for you. It looks like the Hub had been rejecting the messages you were sending for whatever reason, but it seems you have sorted things out yourself.

I will close this issue now, and if it re-emerges, please open another issue.
